088-rebalance-progress-status.md

```diff
@@ -0,0 +1,334 @@
+# Partition Rebalance Progress Status
+
+This proposal introduces a new feature to monitor the progression of an ongoing partition rebalance executed by a Strimzi-managed Cruise Control instance via a `KafkaRebalance` custom resource. 
+
+## Current Situation
+
+At this time, Strimzi users are able to execute partition rebalances via `KafkaRebalance` custom resources but can only monitor the progression of those partition rebalances in two ways:
+
+- Manually querying the Cruise Control REST API endpoint directly. 
+- Inspecting the logs of the Cruise Control instance. 
+
+Unfortunately, neither of these methods are particularly user-friendly.
+
+## Motivation
+
+Information surrounding the progress of an executing partition rebalance is useful for planning future cluster operations. 
+Knowing things like how much time an ongoing partition rebalance has left to take and how much data an ongoing partition rebalance has left to transfer helps users understand the cost of an ongoing partition rebalance.
+This information helps users decide whether they should continue or cancel an ongoing rebalance, and know when future operations will be able to be safely executed.
+
+Further, having this information readily available and easily accessible via Kubernetes primitives, allows users and third-party tools like the Kubernetes CLI or Strimzi Console to easily track the progression of a partition rebalance.
+
+## Proposal
+
+This proposal extends the status section of the `KafkaRebalance` custom resource to include a `progress` section with a nested `rebalanceProgressConfigMap` field that references a `ConfigMap` that contains information related to an ongoing partition rebalance.
```

Comment:
Do we need a new CM? Can't we use the existing one from the proposal?

Reply:
We were considering using the existing ` ConfigMap`, "afterBeforeLoadConfigMap" to store this progress information, but were concerned the additional data would contribute to hitting the 1 MB `ConfigMap` limit sooner. It is not so much of an issue for the constant "non-verbose" executor state information we plan on providing as part of this proposal. However, if we were to extend the feature to provide the variable "verbose" executor state information in the future, it would increase the chance of hitting the limit for larger production clusters that have a larger number of brokers and partitions.

If we have no requests/plans for providing "verbose" executor state information in the future, I don't see much of a problem of storing the information in the existing `ConfigMap`. At the least, it would simplify the proposal implementation . Any thoughts @ppatierno @tomncooper?

Reply:
As mentioned before, I am not sure we'll never have the support for "verbose" so maybe we could think about the present and not the future. From this perspective, using the same ConfigMap seems to be reasonable.
Anyway you can keep the `rebalanceProgressConfigMap` field pointing to that ConfigMap.
When/if one day we have support for "verbose", that field will just point to a different ConfigMap. It should not be a big issue for users.

Reply:
You don't have to use the same CM just because I asked about it.

Reply:
No I just think it's a good compromise. Anyway let's see what @kyguy @tomncooper think?

Reply:
Discussed offline with Paolo and Tom,  since the progress information is constant, we can safely add it to the existing `ConfigMap` maintained for and tied to the `KafkadRebalance` resource. This keeps `KafkaRebalance` information organized in one place, simplifies the proposal implementation, and has insignificant impact on the storage of the `ConfigMap`. Refactored and added this note to the proposal.